app/models/order_transaction.rb

```diff
@@ -0,0 +1,14 @@
+class OrderTransaction < ApplicationRecord
+  belongs_to :order
+  belongs_to :address, foreign_key: 'shipping_address_id'
+
+  enum status: [ :unpaid, :refunded, :paid ]
+  
+  after_create_commit :send_placement_confirmation_mail, if: ->(transaction) { transaction.paid? }
+
+  scope :latest_transaction, ->(order_id) { where(order_id: order_id).order(created_at: :desc).first }
```

Comment:
scope should always return a collection

Reply:
Removed scope and created a class method instead